modules/imaging_browser/php/imagingbrowserrowprovisioner.class.inc

```diff
@@ -0,0 +1,275 @@
+<?php declare(strict_types=1);
+/**
+ * This file implements a data provisioner to get all possible rows
+ * for the imaging browser menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\imaging_browser;
+
+/**
+ * This class implements a data provisioner to get all possible rows
+ * for the imaging browser menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class ImagingBrowserRowProvisioner extends \LORIS\Data\Provisioners\DBRowProvisioner
+{
+    /**
+     * Create an ImagingBrowserRowProvisioner, which gets rows for
+     * the imaging browser menu table.
+     */
+    function __construct()
+    {
+        $config = \NDB_Config::singleton();
+        $user   = \User::singleton();
+        $DB     = \Database::singleton();
+
+
+        // ===========================================================
+        // Get the different scan types to be displayed in data table
+        // ===========================================================
+
+        // Grep all the different scan types present in mri_scan_type
+        $toTable_scan_types = $config->getSetting('tblScanTypes');
+        $all_scan_types_2d  = $DB->pselect(
+            "SELECT ID, Scan_type FROM mri_scan_type mri",
+            array()
+        );
```

Comment:
It seems like it would make more sense to have the list of scan types injected as a parameter to the constructor, rather than querying the database directly here for testability..

@xlecours @ridz1208 (maybe @AlexandraLivadas) thoughts?

Reply:
done.